README.md

```diff
@@ -1,7 +1,124 @@
+</p>
 
-Dialogue is an experimental client-side framework for HTTP-based RPC. The API is influenced by gRPC's Java library.
-Dialogue was designed as a codegen target for [Conjure](https://palantir.github.io/conjure).
+# Dialogue ![Bintray](https://img.shields.io/bintray/v/palantir/releases/dialogue.svg) [![License](https://img.shields.io/badge/License-Apache%202.0-lightgrey.svg)](https://opensource.org/licenses/Apache-2.0)
+
+_Dialogue is a client-side library for HTTP-based RPC, designed to work well with [Conjure](https://palantir.github.io/conjure)-defined APIs._
+
+## Features
+
+- **ConcurrencyLimiters**: additive increase multiplicative decrease (AIMD) concurrency limiters ensure bursty traffic doesn't overload upstream servers.
+- **Client-side node selection**: by making load balancing decisions in the client, Dialogue avoids the necessity for an L7 proxy (and its associated latency penalty).
+- **Queue**: in the case where all nodes are limited (e.g. during a spike in traffic), requests are added to a FIFO queue and processed as soon as the one of the ConcurrencyLimiters has capacity.
+- **Retries**: requests are retried a constant number of times, unless they were streamed.
+- **Live reloading**: uris can be added or removed without losing ConcurrencyLimiter or node selection states.
+- **Content decoding**: JSON, [SMILE](https://github.com/FasterXML/jackson-dataformats-binary/tree/master/smile) and [CBOR](https://github.com/FasterXML/jackson-dataformats-binary/tree/master/cbor) are supported by default, with user-defined encodings also supported.
+- **Streaming**: requests and responses are streamed without buffering the entire body into memory.
+
+## Observability
+
+- **Zipkin-style tracing**: internal operations are instrumented using Zipkin-style [tracing-java spans](https://github.com/palantir/tracing-java), and `X-B3-TraceId` headers are propagated
+- **Metrics**: Timers, meters and gauges are defined using [metric-schema](https://github.com/palantir/dialogue/blob/develop/dialogue-core/src/main/metrics/dialogue-core-metrics.yml) and stored in a [Tritium TaggedMetricRegistry](https://github.com/palantir/tritium).
+- **Structured logging**: SLF4J logs are designed to be rendered as JSON, with every parameter declaratively named.
+
+## Usage
+
+Dialogue works best with Conjure-generated client bindings, i.e. for a given Conjure-defined `FooService`, the [conjure-java](https://github.com/palantir/conjure-java) code generator produces two java interfaces: `FooServiceBlocking` and `FooServiceAsync`. See the [conjure-java generated client bindings][] section below for more details.
+
+**Production usage**: your server framework should provide an abstraction to create clients that handle uri live-reloading and reuse connection pools. For example in Witchcraft, you can create a `FooServiceBlocking` like so:
+
+```groovy
```

Comment:
Curious why you selected groovy over java :-)

Reply:
Declaring it as java gives you pretty much nothing unless its inside some curlies (e.g. a class or function).  Declaring groovy gets you some highlighting straight away.